src/init/starship.zsh

```diff
@@ -89,7 +90,7 @@ VIRTUAL_ENV_DISABLE_PROMPT=1
 
 setopt promptsubst
 
-PROMPT='$(::STARSHIP:: prompt --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")'
-RPROMPT='$(::STARSHIP:: prompt --right --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT")'
+PROMPT='${//$((STARSHIP_REDRAW_COUNT++))}$(::STARSHIP:: prompt --terminal-width="$COLUMNS" --keymap="${KEYMAP:-}" --status="$STARSHIP_CMD_STATUS" --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --cmd-duration="${STARSHIP_DURATION:-}" --jobs="$STARSHIP_JOBS_COUNT" --redraw-count="$STARSHIP_REDRAW_COUNT")'
```

Comment:
Please change this so that an empty command line does not set `STARSHIP_REDRAW_COUNT` to 0, so this information can be used elsewhere more easily. Though even without this change, zsh already seems to have some logic for handling this case.

Maybe also integrate `$((++STARSHIP_REDRAW_COUNT))}` directly into the starship prompt invocation.

Reply:
I do not seem to understand your first suggestion. What do you mean by an empty command line?

As for your second suggestion, if I change it like this:
`--redraw-count="$((++STARSHIP_REDRAW_COUNT))"`
then it will be executed inside of subshell and variable's state won't persist.

Reply:
I would prefer if it also incremented the value instead of resetting to zero for empty inputs in the prompt line that don't execute anything, instead of just vi-mode switches.
This sort of thing:
```
 ❯ 
 ❯ 
 ❯ 
```
As far as I can tell, they only don't send a notification because the `STARSHIP_DURATION` variable is unset just before `STARSHIP_REDRAW_COUNT` is reset to 0.

It has been requested to reset to set the displayed status (and duration) to 0 in such cases, though that approach may not work well with any vi-mode redraw. Unlike the zsh redraw-count approach, your fish init code does increment `redraw-count` for empty lines instead of resetting it.

Reply:
Sorry for disappearing for a couple of months - got busy and totally forgot about this PR. It looks like I can just move `STARSHIP_REDRAW_COUNT=0` to `preexec` instead of `precmd` and it will work as you described.